### PR TITLE
[DO NOT MERGE] Pump frames during invocation of `NativeAutomator` methods

### LINKED ISSUE #1037: App UI gets stuck when executing native actions
**Describe the bug**

When a native method is called (e.g `swipe()` or `enterText()`, app UI gets stuck (technically: `pump()` is not called) when executing native actions, which looks bizarre and confuses the developer.

**To Reproduce**

Run `integration-test/test_test.dart` in [Julia's fork](https://github.com/jBorkowska/flutter-wonderous-app). cc @jBorkowska 

Demo:

https://user-images.githubusercontent.com/40357511/222913229-dd60bbb7-7040-4105-af88-40ec9746b11f.mov

You can see that during swipes, the app's UI is stuck (no frames are pumped).

PS Ignore the long gray splash screen.

**Expected behavior**

App UI is `pump`ed when native actions are executed.

<details>
<summary>Additional information</summary>

```console
$ flutter --version
Flutter 3.7.6 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 12cb4eb7a0 (3 days ago) • 2023-03-01 10:29:26 -0800
Engine • revision ada363ee93
Tools • Dart 2.19.3 • DevTools 2.20.1
```

```
patrol: ^1.0.5
```

```
$ patrol --version                                                                                                                                                                                        
patrol_cli v1.0.5
```

</details>

### PR DESCRIPTION
**Pull request by bartekpacia**
Fixes #1037

Demo from the #1037, with changes made this PR:

https://user-images.githubusercontent.com/40357511/222918260-0d57031e-f385-49ac-a765-58236d287d3c.mov

### CODE REVIEW

File: packages/patrol/lib/src/native/native_automator.dart [hunk 1]
```diff
@@ -689,41 +748,87 @@ class NativeAutomator {
...
+      done.value = true;
+      await done.future;
+    }
+  }
+
+  _Wrap<bool> _startPumping() {
+    final done = _Wrap(false);
+    unawaited(_pump(done));
+    return done;
   }
```
Comment (shilangyu) [inline packages/patrol/lib/src/native/native_automator.dart:812]: Maybe something like:

```dart
Future<T> _pumping<T>(Future<T> Function() action) async {
  final tester = _config.patrolTester;
  final completer = Completer<T>();

  unawaited(
    action().then(completer.complete).catchError(completer.completeError),
  );

  while (!completer.isCompleted) {
    await tester?.pump();
  }

  return completer.future;
}
```

Comment (bartekpacia) [reply packages/patrol/lib/src/native/native_automator.dart]: much better, thanks

**Review by shilangyu — APPROVED** [review]

**Comment by bartekpacia** [comment]
Unfortunately, this breaks when the app under test is not in foreground. The pumping never completes, and we wait forever.

**Comment by bartekpacia** [comment]
By default (when running integration tests on device), `pump()` calls [SchedulerBinding.scheduleFrame()](https://api.flutter.dev/flutter/scheduler/SchedulerBinding/scheduleFrame.html).

We could try using [SchedulerBinding.scheduleForcedFrame()](https://api.flutter.dev/flutter/scheduler/SchedulerBinding/scheduleForcedFrame.html).

**Pull request closed without merging**